bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedHandler.java

```diff
@@ -0,0 +1,456 @@
+                break;
+            case CHANNEL_INTENSITY:
+                if (OnOffType.OFF.equals(command)) {
+                    bigTemp = new BigDecimal(0);
+                } else if (OnOffType.ON.equals(command)) {
+                    bigTemp = new BigDecimal(255);
+                } else {
+                    bigTemp = new BigDecimal(command.toString()).multiply(new BigDecimal(2.55));
+                }
+                sendGetRequest("/win&IX=" + bigTemp);
+                break;
+            case CHANNEL_SLEEP:
+                if (OnOffType.ON.equals(command)) {
+                    sendGetRequest("/win&ND");
+                } else {
+                    sendGetRequest("/win&NL=0");
+                }
+                break;
+            case CHANNEL_PRESETS:
+                sendGetRequest("/win&PL=" + command);
+                break;
+            case CHANNEL_PRESET_DURATION:
+                if (OnOffType.OFF.equals(command)) {
+                    bigTemp = new BigDecimal(0);
+                } else if (OnOffType.ON.equals(command)) {
+                    bigTemp = new BigDecimal(255);
+                } else {
+                    bigTemp = new BigDecimal(command.toString()).multiply(new BigDecimal(600)).add(new BigDecimal(500));
+                }
+                sendGetRequest("/win&PT=" + bigTemp);
+                break;
+            case CHANNEL_TRANS_TIME:
+                if (OnOffType.OFF.equals(command)) {
+                    bigTemp = new BigDecimal(0);
+                } else if (OnOffType.ON.equals(command)) {
+                    bigTemp = new BigDecimal(255);
+                } else {
+                    bigTemp = new BigDecimal(command.toString()).multiply(new BigDecimal(600)).add(new BigDecimal(500));
+                }
+                sendGetRequest("/win&TT=" + bigTemp);
+                break;
+            case CHANNEL_PRESET_CYCLE:
+                if (OnOffType.ON.equals(command)) {
+                    sendGetRequest("/win&CY=1");
+                } else {
+                    sendGetRequest("/win&CY=0");
+                }
+                break;
+        }
+    }
+
+    public void savePreset(int presetIndex) {
+        if (presetIndex > 16) {
+            logger.warn("Presets above 16 do not exist, and the action sent {}", presetIndex);
+            return;
+        }
+        sendGetRequest("/win&PS=" + presetIndex);
+    }
+
+    private void pollLED() {
+        sendGetRequest("/win");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(WLedConfiguration.class);
+        if (!config.address.contains("://")) {
+            logger.debug("Address was not entered in correct format, it may be the raw IP so adding http:// to start");
+            config.address = "http://" + config.address;
+        }
+        pollingFuture = threadPool.scheduleWithFixedDelay(this::pollLED, 1, config.pollTime, TimeUnit.SECONDS);
```

Comment:
The ThingHandler provides an ExecutorService you can use.
```suggestion
        pollingFuture = scheduler.scheduleWithFixedDelay(this::pollLED, 1, config.pollTime, TimeUnit.SECONDS);
```